producers/mimosa26_sbg/src/M26Producer.cxx

```diff
@@ -0,0 +1,511 @@
+  m_running(false), m_triggering(false), m_stopping(false) {
+    m_sumEnDetectors = 0;
+  }
+
+//void reset_com(SInt32 &int1, SInt32 &int2, SInt32 &int3, SInt32 &int4, SInt32 &int5, std::string &cmd) {
+void M26Producer::reset_com() {
+  ret = -1;
+  DaqAnswer_CmdReceived= -1;
+  DaqAnswer_CmdExecuted = -1;
+  VLastCmdError = -1;
+  CmdNumber = -1;
+  cmd = "";
+}
+
+void sendError(std::string cmd){
+  EUDAQ_ERROR(std::string("Send/Receive " + cmd + " failed")); 
+  throw irc::SendReceiveException(std::string("Failed to send/receive "+cmd));
+}
+
+void executeError(std::string cmd){
+  EUDAQ_ERROR(std::string("Execution of " + cmd + " failed")); 
+  throw irc::ExecuteException(std::string("Failed to execute "+cmd));
+}
+
+void M26Producer::ReadoutLoop() {
+  // loop until terminated
+  do {
+    // if not running, move this thread to end of scheduler queue
+    if (!m_running) {
+      sched_yield();
+      continue;
+    } else {
+
+      //try to read shrd mem
+      try{
+	std::lock_guard<std::mutex> lck(m_mutex);
+	// check an read
+	std::cout << " reading" << std::endl;
+
+	// structure of this part depends on how data is send from SBG LV DAQ
+	// - likely to be package with 4 frames per trigger -> 4*n frames = 1 acquisition
+	// ->Is "4" configurable?
+	// - What of more than 1 trigger occurred during "4" frames
+	// -> Data dublication? Does the FW propagate trigger into into RAM such that we know which frames to dublicate
+	// -> Does the MAxTrigNumber influence the FW: longer BUSY? if not, the TLU does not know about this and will continue to issue triggers. 
+	//    Hence LVDAQ might not send data on a certain trigger, but all the other systems are! I think we either need a long busy ("4" frames) or we can not limit the number of triggers so we (need to) know how to send all matching frames to the producer and the producer can dublicate them in such a way that it sends the correct package of "4" frames to the Data Collector
+
+	//datalength1 = ni_control->DataTransportClientSocket_ReadLength("priv");
+	std::vector<unsigned char> mimosa_data_0(1);
+	//mimosa_data_0 = ni_control->DataTransportClientSocket_ReadData(datalength1);
+	unsigned char test = 125; 
+	mimosa_data_0.push_back(test);
+
+	//datalength2 = ni_control->DataTransportClientSocket_ReadLength("priv");
+	std::vector<unsigned char> mimosa_data_1(1);
+	//mimosa_data_1 = ni_control->DataTransportClientSocket_ReadData(datalength2);
+	test = 127; 
+	mimosa_data_1.push_back(test);
+
+
+	eudaq::RawDataEvent ev("NI", m_run, m_ev++);
+	ev.AddBlock(0, mimosa_data_0);
+	ev.AddBlock(1, mimosa_data_1);
+	SendEvent(ev);
+	eudaq::mSleep(283);
+	continue;
+
+      } catch (irc::DataNoEvent &) {
+	// no event in shrd mem
+	sched_yield();
+      }
+
+    }
+
+  } while (!m_terminated);
+}
+
+void M26Producer::OnConfigure(const eudaq::Configuration &param) {
+
+  m_config = param;
+
+  std::cout << "Configuring ...(" << m_config.Name() << ")" << std::endl;

```

Comment:
you do not guard your OnConfig with the mutex, so its call in MainLoop is useless.